providers/efa/verbs.c

```diff
@@ -1118,6 +1118,24 @@ static void efa_sq_advance_post_idx(struct efa_qp *qp)
 		qp->sq.wq.phase++;
 }
 
+/* Don't call directly */
+static inline void efa_wq_ring_doorbell(struct efa_wq *wq, uint16_t desc_idx)
+{
+	mmio_write32(wq->db, desc_idx);
+}
+
+static inline void efa_rq_ring_doorbell(struct efa_rq *rq, uint16_t desc_idx)
+{
+	udma_to_device_barrier();
+	efa_wq_ring_doorbell(&rq->wq, desc_idx);
+}
+
+static inline void efa_sq_ring_doorbell(struct efa_sq *sq, uint16_t desc_idx)
+{
+	mmio_flush_writes();
+	efa_wq_ring_doorbell(&sq->wq, desc_idx);
+}
+
```

Comment:
Let's do not add extra obfuscation. You added more lines than deleted and it is clear indication to me that extra functions are not needed.

Reply:
I don't think the primitive of ringing the doorbell should be hard coded more than once in different flows.
What do you think about dropping efa_wq_ring_doorbell and use the mmio_write32 directly in the RQ/SQ functions?

Reply:
If you can't leave without it, let's do it.

Reply:
Done